README.rst

```diff
@@ -5,52 +5,132 @@ plone.staticresources
+for optimal integration and reuse within Plone frontend ecosystem
+
+
+Themers
+^^^^^^^
+
+Who want to change the look and feel of a Plone site and need to:
+
+- understand how to correctly bundle changes made into existing or new frontend
+code
+-
+
+
+Plone Core Developers
+^^^^^^^^^^^^^^^^^^^^^
+
+Who want to want to help add/enhance this add-on itself.
 
-How to develop on patterns
---------------------------
 
-All JavaScript code in this package is downloaded via yarn into ``src/plone/staticresources/static/components``.
-Nothing in this directory should be touched.
-If you need to fix something, fix it in the original repository and eventually upgrade the version (described below).
+Workflow
+--------
+
+Default Plone buildout configuration adds an executable in
+`./bin/plone-compile-resources` which generates compiled bundles (normal,
+minified and respective maps) in `plone/staticresources/static/`.
+
+.. note::
+
+  You can see all the options of this executable by running
+  ``./bin/plone-compile-resources --help``
+
+The files in this directory are served by Plone. In production mode URLs will
+be similar to:
+
+- http://localhost:8080//++plone++static/++unique++2019-01-08%2006%3A53%3A49.000248/plone-compiled.min.js
+- http://localhost:8080/++plone++static/++unique++2019-01-08%2006%3A53%3A49.040248/plone-logged-in-compiled.min.js
+
+In development mode, XXX: Explain what happens/entry point roles of:
+
+- http://localhost:8080/++resource++plone.js
+- http://localhost:8080/++resource++plone-logged-in.js
+- http://demo.plone.org/++plone++static/plone.less
+- http://localhost:8080//++plone++static/plone-logged-in.less
+
+
+How to develop on ``patterns``
+------------------------------
+
+All JavaScript code in this package is downloaded via ``yarn`` into
+``src/plone/staticresources/static/components``. Nothing in that directory
+should be manually edited. If you need to fix something, do it in the original
+repository and eventually upgrade its version (next section).
+
 For Mockup, the original repository is: https://github.com/plone/mockup/
+
 For Patternslib, visit: http://github.com/patternslib/Patterns
 
 
 How to upgrade the resources in this package
 --------------------------------------------
 
-- Increase the npm package versions in package.json.
-- Run ``yarn upgrade``.
-- Run ``./bin/plone-compile-resources -b plone``, ``./bin/plone-compile-resources -b plone-logged-in`` and whatever bundle you're going to build in your buildout's root directory.
-  For more info on ``plone-compile-resources`` see: XXX
-- Increase the ``last_compilation`` date in ``src/plone/staticresources/profiles/default/registry/bundles.xml``.
+- Increase ``npm`` package versions in ``package.json``, i.e. sections
+``dependencies`` or ``devDependencies``.
+- Run ``yarn upgrade`` (cannot be ``npm``)
+- Run ``./bin/plone-compile-resources -b plone``,
+``./bin/plone-compile-resources -b plone-logged-in`` and whatever bundle you're
```

Comment:
./bin/plone-compile-resources compiles them all, right? Can we just suggest to use ./bin/plone-compile-resources and if somebody wants to do a partial compilation he can do that separately?